homeassistant/components/enigma2/config_flow.py

```diff
@@ -115,23 +115,45 @@ async def async_step_user(
         if user_input is None:
             return self.async_show_form(step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA)
 
-        data = await self.validate_user_input(user_input)
+        await self.validate_user_input(user_input)
         if "base" in self.errors:
             return self.async_show_form(
                 step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA, errors=self.errors
             )
         return self.async_create_entry(
-            data=data, title=data[CONF_HOST], options=self._options
+            data=self._data, title=self._data[CONF_HOST], options=self._options
```

Comment:
```suggestion
            data=user_input, title=user_input[CONF_HOST], options=self._options
```
I don't see the need for self._data